### PR TITLE
Implement "Delete for me" support for messages

### PR DESCRIPTION
**Pull request by vurpo**
Partially implemented: redacting messages on Matrix when they are deleted "for me" on WhatsApp. Still missing: checking whether or not I am the only Matrix user in the portal before redacting (they should not be redacted if there are other people).

### CODE REVIEW

**Comment by vurpo** [comment]
@tulir do you have any good ideas on how to check whether there are more Matrix users in the portal? StateStore did not offer help with that, and I did find a method for doing it when casting it to SQLStateStore, but casting down to a type feels somehow wrong.

**Comment by tulir** [comment]
Just noticed that `portal.bridge.StateStore` allows accessing it without casting, so I guess either that or just fetch the member list every time. It can be optimized later, I don't think people are constantly using the delete for me feature.

File: portal.go [hunk 1]
```diff
@@ -1664,6 +1664,28 @@ func (portal *Portal) HandleMessageRevoke(user *User, info *types.MessageInfo, k
 	return true
 }
 
+func (portal *Portal) deleteForMe(user *User, content *events.DeleteForMe) bool {
+	matrixUsers, err := portal.GetMatrixUsers()
+	if err != nil {
+		portal.log.Errorfln("Unable to get Matrix users for portal, to see if DeleteForMe should be handled or not")
```
Comment (tulir) [inline portal.go:1670]: This should actually log the error (also adjusted the error message slightly to match the usual style)

```suggestion
		portal.log.Errorln("Failed to get Matrix users in portal to see if DeleteForMe should be handled:", err)
```

**Review by tulir — APPROVED** [review]
Error logs should log the error itself, otherwise looks good

**Pull request merged**